pkg/runtime/runtime.go

```diff
@@ -585,19 +595,27 @@ func (a *DaprRuntime) sendBindingEventToApp(bindingName string, data []byte, met
 		req.WithMetadata(reqMetadata)
 
 		ctx := context.Background()
-		spanName := fmt.Sprintf("Binding: %s", bindingName)
-		// context.Background() can be considered as GRPC context and so using StartTracingServerSpanFromGRPCContext to generate span
-		ctx, span := diag.StartTracingServerSpanFromGRPCContext(ctx, spanName, a.globalConfig.Spec.TracingSpec)
-		defer span.End()
-
-		ctx = diag.NewContext(ctx, span.SpanContext())
+		isTracingEnabled := diag_utils.IsTracingEnabled(a.globalConfig.Spec.TracingSpec.SamplingRate)
+		var spanName string
+		var span *trace.Span
+		// start and update the trace span only when tracing is enabled - sampling rate is non zero
+		if isTracingEnabled {
+			spanName = fmt.Sprintf("Binding: %s", bindingName)
```

Comment:
Can we move all tracing code to invokeMethod of http appchannel?

Reply:
I didnt get this completely , why do we want to move to invokeMethod?

Reply:
bindings and publish codes for http appchannel are very similar, once we move this tracing code to invokemethod. we can remove the duplicated code. but, yes. we need to differentiated between two sources. I think we can set context value to do it.